common/src/format.rs

```diff
@@ -433,7 +460,42 @@ impl FormatSpec {
                 let ch = char::from(self.format_type.as_ref().unwrap());
                 Err(FormatSpecError::UnknownFormatCode(ch, "float"))
             }
-            Some(FormatType::Number) => Err(FormatSpecError::NotImplemented('n', "float")),
+            Some(FormatType::Number) => {
+                const UPPER_BOUND: f64 = 1e+5;
+                const LOWER_BOUND: f64 = 1e-4;
+                const ZERO_SEP: char = '\x00';
+
+                let show_as_exponential = magnitude >= UPPER_BOUND || magnitude <= LOWER_BOUND;
+                let precision = if precision == 0 { 1 } else { precision };
+                let mut inter = self.get_separator_interval();
+
+                let width = self.width.unwrap_or(precision);
+                let disp_digit_cnt = width;
+
+                let magnitude_str: String = if show_as_exponential {
+                    float_ops::format_general(
+                        precision,
+                        magnitude,
+                        Case::Lower,
+                        self.alternate_form,
+                        true,
+                    )
+                } else {
+                    // TODO: currently, `nan` and `inf` are not work properly. should handle this
+                    if magnitude.is_nan() {
+                        return Ok("nan".to_string());
+                    }
+                    self.check_number_formatting_cases(magnitude, width, precision, &mut inter)
+                };
+
+                Ok(FormatSpec::add_magnitude_separators_for_char(
+                    magnitude_str,
+                    inter as i32,
+                    ZERO_SEP,
+                    disp_digit_cnt as i32,
+                    PaddingStyle::Zero,
```

Comment:
Here's where you should use `PaddingStyle::WhiteSpace`.

Reply:
Oh why didn't I see this?? edited again